### PR TITLE
Recognize symlinks in compilation database and eliminate duplication

### PR DESCRIPTION
**Pull request by Ducnguyen147**
Previously, the compilation database JSON file could potentially include duplicate entries for files that were symlinked. This caused issues when building the project as it would result in multiple compilations of the same file. To solve this problem, script was modified to recognize symlinks and eliminate any duplicate entries in the compilation database. This ensures that each file is only compiled once, reducing build times and avoiding errors.

### CODE REVIEW

**Review by vodorok — CHANGES_REQUESTED** [review]
Hi Ducnguyen147, 
Thank you for your contribution. 
Please see my remarks.
Please modify the help texts in the `cmd/analyze.py` and `cmd/check.py` as this new option is not yet reflected in the help messages of the respective parameters. There is a section in the `docs/analyzer/user_guide.md` too.
Also, I would like to see some tests.

@bruntib  please challenge the comments in this review.

**Comment by vodorok** [comment]
Sorry, my initial review comments were lost, that's why the edits. Keep up the good work! :rocket:

**Comment by Ducnguyen147** [comment]
Sorry I just click on the wrong re-request button. Thank you for your suggestions @vodorok 👏, I’m on your comments

**Comment by vodorok** [comment]
No worries!

File: analyzer/codechecker_analyzer/buildlog/log_parser.py [hunk 1]
```diff
@@ -1312,6 +1328,12 @@ def parse_unique_log(compilation_database,
                 elif action.output <\
                         uniqued_build_actions[action.source].output:
                     uniqued_build_actions[action.source] = action
+            elif build_action_uniqueing == CompileActionUniqueingType.SYMLINK:
+                # Dictionary stores build actions of the unique source files
+                # The non-symlink version will be kept if it exists
+                if (not is_symlink(source_file)) or\
```
[the remark this answers is not in the document]
Comment (bruntib) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:1334]: The example is good, but we'd like to recognize symlink directories too. So this example could be extended by `dirE` which is a symlink to `dirA`.

File: analyzer/codechecker_analyzer/buildlog/log_parser.py [hunk 2]
```diff
@@ -1186,7 +1186,16 @@ class CompileActionUniqueingType(Enum):
...
     SOURCE_REGEX = 2  # Based on source file, uniqueing by regex filter
     STRICT = 3  # Gives error in case of duplicate
+    SYMLINK = 4 # Recognizing symlinks and eliminate the duplications
 
+def is_symlink(filepath):
+    """
+    Checks whether the given file path is symlink or located in the 
+    directory containing a symlink.
+    """
+    parent_dir = os.path.abspath(filepath)
```
Comment (bruntib) [inline analyzer/codechecker_analyzer/buildlog/log_parser.py:1196]: As @vodorok mentioned in another comment, the "directory" and "file" components always need to be handled together. `os.path.abspath()` returns an absolute path of the given file, but the current working directory will be used as prefix.

**Review by bruntib — CHANGES_REQUESTED** [review]
The direction is good, thanks for the patch!

**Comment by Ducnguyen147** [comment]
Dear @vodorok and @bruntib , the log_parser.py file has been edited. You can review it!

**Comment by Ducnguyen147** [comment]
I'm working on the help texts! Thank you!

**Review by bruntib — APPROVED** [review]

**Pull request merged**